tests/components/ring/conftest.py

```diff
@@ -36,6 +38,63 @@ def mock_ring_auth():
         yield mock_ring_auth.return_value
 
 
+@pytest.fixture
+def mock_ring_devices():
+    """Mock Ring devices."""
+
+    devices = get_mock_devices()
+    device_list = list(chain.from_iterable(devices.values()))
+
+    class FakeRingDevices:
+        """Class fakes the RingDevices class."""
+
+        all_devices = device_list
+        video_devices = (
+            devices["stickup_cams"]
+            + devices["doorbots"]
+            + devices["authorized_doorbots"]
+        )
+        stickup_cams = devices["stickup_cams"]
+        other = devices["other"]
+        chimes = devices["chimes"]
+
+        get_device = lambda _, id: [
+            device for device in device_list if device.id == id
+        ][0]
+        get_video_device = lambda _, id: [
+            device
+            for device in device_list
+            if (
+                device.id == id
+                and device.family in {"stickup_cams", "doorbots", "authorized_doorbots"}
+            )
+        ][0]
+        get_stickup_cam = lambda _, id: [
+            device
+            for device in device_list
+            if (device.id == id and device.family == "stickup_cams")
+        ][0]
+        get_other = lambda _, id: [
+            device
+            for device in device_list
+            if (device.id == id and device.family == "others")
+        ][0]
+
+    return FakeRingDevices()
+
+
+@pytest.fixture
+def mock_ring_client(mock_ring_auth, mock_ring_devices):
+    """Mock ring client api."""
+    mock_client = create_autospec(ring_doorbell.Ring)
+    mock_client.return_value.devices_data = get_devices_data()
+    mock_client.return_value.devices.return_value = mock_ring_devices
+    mock_client.return_value.active_alerts.side_effect = get_active_alerts
+
+    with patch("homeassistant.components.ring.Ring", new=mock_client):
```

Comment:
Do we mock the Ring lib used in the config flow somewhere? Maybe we can mock both

Reply:
The `Auth` class is used in the config flow and is mocked here https://github.com/home-assistant/core/blob/dev/tests/components/ring/conftest.py#L28.  It doesn't need to be mocked in `__init__` because it's just instantiated then passed into the `Ring` mock which ignores it.

Reply:
Resolving conversation as I believe this is the answer, that the `Auth` part of the ring lib is mocked in config_flow.